### PR TITLE
BUG: lib.infer_dtype with incompatible intervals

### PR DESCRIPTION
**Pull request by jbrockmendel**
Nothing user-facing.

### CODE REVIEW

File: pandas/_libs/lib.pyx [hunk 1]
```diff
@@ -2035,10 +2035,58 @@ cdef class IntervalValidator(Validator):
...
 
 cpdef bint is_interval_array(ndarray values):
+    """
+    Is this an ndarray of Interval (or np.nan) with a single dtype?
+    """
+
     cdef:
-        IntervalValidator validator = IntervalValidator(len(values),
-                                                        skipna=True)
-    return validator.validate(values)
```
Comment (jreback) [inline pandas/_libs/lib.pyx:2041]: can you remove the IntervalValidator code?

Comment (jbrockmendel) [reply pandas/_libs/lib.pyx:2041]: sure, will add that to the upcoming misc branch

Comment (jreback) [reply pandas/_libs/lib.pyx:2041]: kk

File: pandas/_libs/lib.pyx [hunk 2]
```diff
@@ -2035,10 +2035,58 @@ cdef class IntervalValidator(Validator):
...
+                    return False
+            elif td64:
+                if not is_timedelta(val.left):
+                    return False
+            elif dt64:
+                if not PyDateTime_Check(val.left):
+                    return False
+            else:
+                raise ValueError(val)
+        elif util.is_nan(val) or val is None:
```
Comment (jreback) [inline pandas/_libs/lib.pyx:2081]: couldn't this be a float nan or NA? (or None)? i am not sure we really specify

File: pandas/_libs/lib.pyx [hunk 3]
```diff
@@ -2035,10 +2035,58 @@ cdef class IntervalValidator(Validator):
...
+            pass
+        else:
+            return False
+
+    if closed is None:
+        # we saw all-NaTs, no actual Intervals
```
Comment (jreback) [inline pandas/_libs/lib.pyx:2087]: all-missing / nan?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/_libs/lib.pyx (hunk 2 above)
Comment (jbrockmendel) [reply pandas/_libs/lib.pyx:2081]: this matches what we do in corresponding libinterval function

File: pandas/_libs/lib.pyx (hunk 3 above)
Comment (jbrockmendel) [reply pandas/_libs/lib.pyx:2087]: yah, will update in the follow-on brnach

File: pandas/_libs/lib.pyx (hunk 2 above)
Comment (jreback) [reply pandas/_libs/lib.pyx:2081]: got it
ok i think we should open an issue about this (eg letting the missing value be NA) or NaT if it's a datetimelike in a new issue as this is a bit inconsistent with what we allow for missing values in other dtypes 

but out of scope for this PR

File: pandas/_libs/lib.pyx (hunk 3 above)
Comment (jbrockmendel) [reply pandas/_libs/lib.pyx]: updated + green

**Review by jreback — APPROVED** [review]

**Pull request merged**